app/src/app.js

```diff
@@ -272,21 +272,24 @@
                 var name = $('.new-collection-name').val(),
                     isPublic = $('.new-collection-public').is(':checked');
 
-                girder.restRequest({
-                    path: 'collection/romanesco/module',
-                    type: 'post',
-                    data: {
-                        name: name,
-                        description: '',
-                        public: isPublic
-                    }
-                }).done(_.bind(function (collection) {
+                var collection = new girder.models.CollectionModel({
+                    name: name,
+                    public: isPublic
+                }).on('g:saved', function () {
                     $('.new-collection-name').val('');
                     this.collection.fetch({}, true);
-                }, this)).error(_.bind(function (xhr, status, message) {
+
+                    _.each(['Data', 'Analyses', 'Visualizations'], function (name) {
+                        new girder.models.FolderModel({
+                            parentType: 'collection',
+                            parentId: collection.get('_id'),
+                            name: name
+                        }).save();
```

Comment:
@zachmullen the folders need to be public based on isPublic also. Right now they are always private, even if the collection is public.